File: src/main/kotlin/com/zoom/webapp/springguruwebapp/services/BookServiceImpl.java
package com.zoom.webapp.springguruwebapp.services;

import com.zoom.webapp.springguruwebapp.domain.Book;
import com.zoom.webapp.springguruwebapp.repositories.BookRepository;
import org.springframework.stereotype.Service;

@Service
public class BookServiceImpl implements BookService {

    private final BookRepository bookRepository;

    public BookServiceImpl(BookRepository bookRepository) {
        this.bookRepository = bookRepository;
    }

    @Override
    public Iterable<Book> findAll() {
        return this.bookRepository.findAll();
    }
}

File: spring-webapp/src/main/kotlin/com/zoom/webapp/springguruwebapp/services/BookServiceImpl.kt
package com.zoom.webapp.springguruwebapp.services

import com.zoom.webapp.springguruwebapp.domain.Book
import com.zoom.webapp.springguruwebapp.repositories.BookRepository
import org.springframework.stereotype.Service

@Service
class BookServiceImpl(
    private val bookRepository: BookRepository
) : BookService {
    override fun findAll(): Iterable<Book> {
        return bookRepository.findAll()
    }
}